service/bookturks/quiz/QuizTools.py

```diff
@@ -238,3 +239,21 @@ def unlink_all_tags_of_quiz(quiz):
         """
         for tag in quiz.quiztag_set.all():
             QuizTagAdapter.unlink_quiz(tag_name=tag.tag_name, quiz_id=quiz.quiz_id)
+
+    @staticmethod
+    def check_attempt_eligibility(self, request, quiz_complete_model, quiz_id):
```

Comment:
Try not to pass the whole request object as a parameter.
The idea is to narrow down the parameters to only what is required.
The request object contains a lot of unnecessary stuff with it.
It also causes a lot of trouble while writing test cases for this function. Its hard to replicate this object to test it.
So I would suggest just pass the model which needs to be checked.